### PR TITLE
Update ExportJob so that it can be used to construct/run Dataflow template

### PR DESCRIPTION
**Pull request by kevinsi4508**

### CODE REVIEW

File: bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java [hunk 1]
```diff
@@ -37,4 +47,94 @@ public static CloudBigtableTableConfiguration BuildImportConfig(ImportOptions op
     }
     return builder.build();
   }
+
+  /**
+   * Provides a request that is constructed using a {@link Scan} object.
+   */
+  private static class RequestWithScanValueProvider
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java]: Scan is an internal implementation detail, it just takes attributes for a ReadRowsRequest and builds it. I don't think RequestWithScan makes sense as a name

File: bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java [hunk 2]
```diff
@@ -37,4 +47,94 @@ public static CloudBigtableTableConfiguration BuildImportConfig(ImportOptions op
...
+            opts.getBigtableMaxVersions(),
+            opts.getBigtableFilter());
+    CloudBigtableScanConfiguration.Builder configBuilder =
+        new CloudBigtableScanConfiguration.Builder()
+            .withProjectId(opts.getBigtableProject())
+            .withInstanceId(opts.getBigtableInstanceId())
+            .withTableId(opts.getBigtableTableId())
+            .withRequest(request);
+
+    if (opts.getBigtableAppProfileId() != null) {
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java]: app profile id is a value provider, the null check doesn't make sense

File: bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java (hunk 1 above)
Comment (kevinsi4508) [reply bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java]: Yeah, I have doubt about the naming.
When you say "attributes", are you referring to following?
+    private final ValueProvider<String> start;
+    private final ValueProvider<String> stop;
+    private final ValueProvider<Integer> maxVersion;
+    private final ValueProvider<String> filter;

I will just call this class RequestValueProvider then.

Let me know.

File: bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java (hunk 2 above)
Comment (kevinsi4508) [reply bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java]: Yeap, I was going to chain them up and without checking null, I will add withAppProfileId()
 to CloudBigtableTableConfiguration.Builder and CloudBigtableScanConfiguration.Builder first.

Comment (kevinsi4508) [reply bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java]: updated/merged.

File: bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java [hunk 3]
```diff
@@ -37,4 +47,89 @@ public static CloudBigtableTableConfiguration BuildImportConfig(ImportOptions op
...
+      this.maxVersion = maxVersion;
+      this.filter = filter;
+    }
+
+    @Override
+    public ReadRowsRequest get() {
+      if (cachedRequest == null) {
+        Scan scan = new Scan();
+        if (start.get() != null && !start.get().isEmpty()) {
+          scan.setStartRow(stop.get().getBytes());
```
[the remark this answers is not in the document]
Comment (kevinsi4508) [reply bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java]: Good catch! thanks!

File: bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java [hunk 4]
```diff
@@ -37,4 +47,89 @@ public static CloudBigtableTableConfiguration BuildImportConfig(ImportOptions op
...
+          scan.setStopRow(stop.get().getBytes());
+        }
+        if (maxVersion.get() != null) {
+          scan.setMaxVersions(maxVersion.get());
+        }
+        if (filter.get() != null && !filter.get().isEmpty()) {
+          try {
+            scan.setFilter(new ParseFilter().parseFilterString(filter.get()));
+          } catch (CharacterCodingException e) {
+            e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (kevinsi4508) [reply bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/TemplateUtils.java]: done.

**Comment by kevinsi4508** [comment]
PTAL

**Pull request merged**